pkg/pillar/zedcloud/authen.go

```diff
@@ -454,26 +479,37 @@ func VerifySigningCertChain(ctx *ZedCloudContext, content []byte) ([]byte, error
 		Intermediates: interm,
 	}
 	if _, err := leafcert.Verify(opts); err != nil {
-		log.Errorf("VerifySigningCertChain: fail to verify the cert chain %v\n", err)
-		return nil, err
+		errStr := fmt.Sprintf("signature verification fail, %v\n", err)
+		log.Errorf("verifySignature: " + errStr)
+		return errors.New(errStr)
 	}
+	return nil
+}
+
+func UpdateServerCert(ctx *ZedCloudContext, certByte []byte) error {
 
-	// save this cert and hash to serverSigningCert and serverSigingCertHash
+	// store the certificate hash
 	ctx.serverSigningCertHash = ComputeSha(certByte)
```

Comment:
Shouldn't we defer this until the function has succeeded?

Reply:
sure